### PR TITLE
mergeStyle[Sets]WithOptions accepts options

### PR DESCRIPTION
**Pull request by jdhuntington**
- Accepts an RTL boolean to specify direction
 - Pulls RTL access into a single location, removes dependency on state

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10180)

### CODE REVIEW

File: packages/merge-styles/src/StyleOptionsState.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { IStyleOptions } from './IStyleOptions';
+
+/**
+ * Sets the current RTL value.
+ */
+export function setRTL(isRTL: boolean): void {
```
Comment (dzearing) [inline packages/merge-styles/src/StyleOptionsState.ts]: should this be `rtl.ts`?

Comment (jdhuntington) [reply packages/merge-styles/src/StyleOptionsState.ts]: I anticipate this file gathering all "global" state for stylesheets. Another we've talked about adding is a per-component stylesheet in the case of web components.

File: packages/merge-styles/src/styleToClassName.test.ts [hunk 2]
```diff
@@ -14,293 +11,348 @@ describe('styleToClassName', () => {
...
 
     expect(_stylesheet.getRules()).toEqual('.css-0 .foo{background:red;}');
   });
 
   it('can have child selectors with comma', () => {
-    styleToClassName({
-      selectors: {
-        '.foo, .bar': { background: 'red' }
+    styleToClassName(
+      {},
```
Comment (aneeshack4) [inline packages/merge-styles/src/styleToClassName.test.ts]: Is this just a different way to format the selectors (the { {}, {selectors : {....}}}?

File: packages/merge-styles/src/extractStyleParts.ts [hunk 3]
```diff
@@ -1,48 +1,54 @@
...
 /**
  * Separates the classes and style objects. Any classes that are pre-registered
  * args are auto expanded into objects.
  */
-export function extractStyleParts(
-  ...args: (IStyle | IStyle[] | false | null | undefined)[]
-): { classes: string[]; objects: IStyleBaseArray } {
-  const classes: string[] = [];
-  const objects: {}[] = [];
+export function extractStyleParts(...args: (IStyle | IStyle[] | false | null | undefined)[]): IExtractStylePartsResult {
```
Comment (aneeshack4) [inline packages/merge-styles/src/extractStyleParts.ts]: Should we also write tests for this?

Comment (jdhuntington) [reply packages/merge-styles/src/extractStyleParts.ts]: Would be ideal, yes. In this case this was just moving code and not writing anything new.

File: packages/merge-styles/src/styleToClassName.test.ts (hunk 2 above)
Comment (jdhuntington) [reply packages/merge-styles/src/styleToClassName.test.ts]: I added a new param to `styleToClassName`, and this is just calling it the same way it had been.

**Comment by dzearing** [comment]
curious why large the jump in bytes

File: packages/merge-styles/src/mergeStyleSetsWithOptions.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+ *
+ * @param options - Set of options to be used when calculating styles.
+ * @param styleSets - One or more style sets to be merged.
+ */
+export function mergeStyleSetsWithOptions(
+  options: IStyleOptions,
+  ...styleSets: Array<IStyleSet<any> | undefined | false | null>
+): IProcessedStyleSet<any>;
+
+export function mergeStyleSetsWithOptions<TStyleSet extends IStyleSet<TStyleSet>>(
```
Comment (dzearing) [inline packages/merge-styles/src/mergeStyleSetsWithOptions.ts]: is this dupe code? I'm trying to understand why the huge bundle regression.

**Pull request closed without merging**